File: 1-TwoSum/1-TwoSum.py
# Last updated: 22/2/2026, 10:20:43 pm
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # Pair each number with its original index
        nums_with_index = [(num, i) for i, num in enumerate(nums)]
        # Sort by value
        nums_with_index.sort()

        l, r = 0, len(nums_with_index) - 1

        while l < r:
            total = nums_with_index[l][0] + nums_with_index[r][0]
            if total == target:
                return [nums_with_index[l][1], nums_with_index[r][1]]
            elif total < target:
                l += 1
            else:
                r -= 1

File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 28/7/2026, 12:43:30 pm
1class Solution {
2public:
3    vector<int> twoSum(vector<int>& nums, int target) {
4        unordered_map<int,int>mp;
5        int diff;
6        for(int i=0;i<nums.size();i++){
7            diff=target-nums[i];
8            if(mp.find(diff)!=mp.end()){
9                return{mp[diff],i};
10            }
11            mp[nums[i]]=i;
12        }
13        return{-1,-1};
14        
15    }
16};